### PR TITLE
Feed HLS and MP4 url to videojs

### PR DESCRIPTION
**Pull request by rmens**
Fetch HLS and MP4 files from the Vimeo API. This improves compatibility with certain android devices that don't support hls. Also it makes Chromecast work.

### CODE REVIEW

**Comment by rmens** [comment]
@marijnvdwerf on old episodes where there is no HD mp4, the site falls back to the Vimeo player. We probably need a bit of 'pick the best quality mp4' logic here. Based on the `type` == `video/mp4` and `width` is highest. Can you take a look at expanding my first try?

```
 "files": [
    {
      "quality": "sd",
      "type": "video/mp4",
      "width": 426,
      "height": 240,
      "link": "xxxx",
      "created_time": "2021-11-01T17:33:45+00:00",
      "fps": 25,
      "size": 112815249,
      "md5": "75aabfa96a423688dd7bfc9739f7924a",
      "public_name": "SD 240p",
      "size_short": "107.59MB"
```

File: functions.php [hunk 1]
```diff
@@ -98,6 +98,18 @@ function zw_filter_pre_oembed_result($default, $url, $args)
         return $default;
     }
 
+    $mp4 = null;
+    foreach ($body->files as $source) {
+        if ($source->quality === 'hd') {
+            $mp4 = $source->link;
+            break;
+        }
```
Comment (marijnvdwerf) [inline functions.php:106]:
```php
        if ($source->type !== "video/mp4") continue;

        if (!$mp4 || $source->width > $mp4->width) {
            $mp4 = $source;
        }
```

**Review by marijnvdwerf — CHANGES_REQUESTED** [review]

File: functions.php [hunk 2]
```diff
@@ -86,18 +86,27 @@ function zw_filter_pre_oembed_result($default, $url, $args)
...
             break;
         }
     }
 
-    if ($m3u === null) {
+    if ($m3u8 === null) {
         return $default;
     }
 
+    $mp4 = null;
```
Comment (marijnvdwerf) [inline functions.php:101]: `var_dump($body->files); die()`?

File: functions.php [hunk 3]
```diff
@@ -86,18 +86,27 @@ function zw_filter_pre_oembed_result($default, $url, $args)
...
+    foreach ($body->files as $source) {
+        if ($source->type !== "video/mp4") continue;
+
```
Comment (marijnvdwerf) [inline functions.php:104]: (add)
```php
if (!isset($mp4->width)) {
    $mp4->width = 0;
}

**Comment by rmens** [comment]
All edge cases handled. No more errors when running with `wp_debug` enabled. Think we are good to merge.

**Review by marijnvdwerf — APPROVED** [review]
Indenting is a bit weird, but should be functional like this

**Pull request merged**